src/ray/protobuf/common.proto

```diff
@@ -422,6 +422,12 @@ message TaskSpec {
   // This will be the actor creation task's task id for concurrent actors. Or
   // the main thread's task id for other cases.
   bytes submitter_task_id = 33;
+  // Some timestamps of the task's lifetime, useful for metrics.
+  // The time the task's dependencies have been resolved.
+  int64 dependency_resolution_timestamp_ms = 34;
```

Comment:
Task spec is supposed to encapsulate all immutable information about the task but we are now adding more and more dynamic stuff to it. Maybe in the future we should separate mutable and immutable into two structs?

Reply:
hmmm yeah i didn't know that task spec was supposed to be immutable but that seems reasonable to me.